frontend/src/views/NftDisplay.vue

```diff
@@ -245,6 +399,90 @@ export default Vue.extend({
       document.execCommand('copy');
       document.body.removeChild(dummy);
     },
+    convertWeiToSkill(wei: string): string {
+      return fromWeiEther(wei);
+    },
+    maxPrecisionSkill(listedPrice: string): string {
+      return this.convertStringToDecimal(
+        this.convertWeiToSkill(listedPrice),
+        8
+      );
+    },
+    convertStringToDecimal(val: string, maxDecimals: number) {
+      return new BigNumber(val).toFixed(maxDecimals);
+    },
+    async purchaseNft() {
+      this.marketOutcome = null;
+      if (this.nftId === null) return;
+
+      const price = await this.lookupNftPrice(this.nftId);
+      if (price) {
+        this.nftPrice = price;
```

Comment:
I need some clarification here. When purchasing we want to check if displayed price equals price on chain. Here you read price from chain and assign it to UI price to later on check if they are the same. This is confusing, because now they will always be the same even though they could be different before you reassigned UI price here.

I think in original Market we just do `if(!price) return;`